scripts/generate_config_yaml.py

```diff
@@ -0,0 +1,131 @@
+    with open(os.path.join(templates_path, 'node_spinning.txt'), 'r') as content_file:
+        content = Template(content_file.read())
+
+    file_output.write('  Node Spinnig Results:\n')
+
+    random_number = rand_x_digit_num(19)
+    file_output.write(content.substitute(ci_name=ci_name,
+                                         random_number=random_number))
+
+
+def overhead_simple_publisher_and_subscriber(file_output, rmw_implementations, ci_name):
+    """Generate yaml file for overhead simple publisher and subscriber tests."""
+    templates_path = os.path.join(get_package_share_directory('buildfarm_perf_tests'), 'templates')
+    add_line_to_plot_overhead(file_output,
+                              os.path.join(templates_path, 'overhead_round_trip.txt'),
+                              'Overhead simple publisher and subscriber - Average Round-Trip Time:',
+                              rmw_implementations, ci_name)
+    add_line_to_plot_overhead(
+                file_output,
+                os.path.join(templates_path, 'overhead_received_packets.txt'),
+                'Overhead simple publisher and subscriber - Received packets per second:',
+                rmw_implementations,
+                ci_name)
+    add_line_to_plot_overhead(file_output,
+                              os.path.join(templates_path, 'overhead_sent_packets.txt'),
+                              'Overhead simple publisher and subscriber - Sent packets per second:',
+                              rmw_implementations, ci_name)
+    add_line_to_plot_overhead(file_output,
+                              os.path.join(templates_path, 'overhead_lost_packets.txt'),
+                              'Overhead simple publisher and subscriber - Lost packets per second:',
+                              rmw_implementations, ci_name)
+    add_line_to_plot_overhead(file_output,
+                              os.path.join(templates_path, 'overhead_virtual_memory.txt'),
+                              'Overhead simple publisher and subscriber - Virtual Memory:',
+                              rmw_implementations, ci_name)
+    add_line_to_plot_overhead(
+                file_output,
+                os.path.join(templates_path, 'overhead_resident_anonymous_memory.txt'),
+                'Overhead simple publisher and subscriber - Resident Anonymous Memory:',
+                rmw_implementations,
+                ci_name)
+    add_line_to_plot_overhead(file_output,
+                              os.path.join(templates_path, 'overhead_physical_memory.txt'),
+                              'Overhead simple publisher and subscriber - Physical Memory:',
+                              rmw_implementations, ci_name)
+
+
+def main():
+    """Call main function."""
+    parser = argparse.ArgumentParser(description='Create ros buildfarm config.')
+    parser.add_argument('-filename', dest='filename', required=True,
```

Comment:
I found no other packages in ROS 2 that use a single dash for full arguments like this. If you want this so that you can have a short form, I'd suggest something like `add_argument('-f', '--filename', ...` instead.

You can also drop the "dest" for all of these - the value you're giving is the same as what the default would be.

Reply:
https://github.com/ros2/buildfarm_perf_tests/pull/29/commits/0596d992ca6c89b26cc78b987ef6aa77ef5ddc8b